.github/actions/build-docker/action.yml

```diff
@@ -0,0 +1,105 @@
+    description: 'The image repository, e.g. camunda/zeebe'
+    default: 'camunda/zeebe'
+    required: true
+  version:
+    description: 'The image version, e.g. SNAPSHOT, 8.1.0'
+    required: false
+  package:
+    description: 'If true, will package Zeebe, otherwise uses the tar-ball under dist/target'
+    default: 'false'
+    required: false
+  push:
+    description: 'If true, will push the image'
+    required: false
+    default: 'false'
+
+outputs:
+  image:
+    description: "Fully qualified image name available in your local Docker daemon"
+    value: ${{ steps.get-image.outputs.result }}
+  date:
+    description: "The ISO 8601 date at which the image was created"
+    value: ${{ steps.get-date.outputs.result }}
+  version:
+    description: "The semantic version of the packaged artifact"
+    value: ${{ steps.get-version.outputs.result }}
+
+runs:
+  using: composite
+  steps:
+    # Creating a context is required when installing buildx on self-hosted runners
+    - name: Create context
+      shell: bash
+      run: |
+        docker context create zeebe-context
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        # to be able to push to local registry, which we use in our tests, we need to use host network
+        driver-opts: network=host
+        endpoint: zeebe-context
+        install: true
+    # We need maven and Java (indirectly) to get properties from the project
+    - uses: actions/setup-java@v3.4.1
+      with:
+        distribution: 'temurin'
+        java-version: '17'
+    - uses: stCarolas/setup-maven@v4.4
+      with:
+        maven-version: 3.8.5
+    - if: ${{ inputs.package == 'true' }}
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: 'clients/go/go.mod'
+        cache: true
+        cache-dependency-path: 'clients/go/go.sum'
+    - if: ${{ inputs.package == 'true' }}
+      name: Build Go
+      shell: bash
+      run: ./build.sh
+      working-directory: clients/go/cmd/zbctl
+    - if: ${{ inputs.package == 'true' }}
+      name: Package Zeebe
+      shell: bash
+      run: mvn -B -DskipTests -DskipChecks package -T1C
+    - name: Set semantic version from Maven project
+      id: get-version
+      shell: bash
+      run: echo ::set-output name=result::$(mvn -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec)
+    - name: Set image build label from ISO 8601 DATE
+      id: get-date
+      shell: bash
+      run: echo "::set-output name=result::$(date --iso-8601=seconds)"
+    - name: Set image name from params or project version
+      id: get-image
+      shell: bash
+      run: echo "::set-output name=result::${{ inputs.repository }}:${{ inputs.version || steps.get-version.outputs.result }}"
+    - name: Build Docker image
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        tags: ${{ steps.get-image.outputs.result }}
+        load: ${{ inputs.push != 'true' }}
```

Comment:
:question: not sure i understand, we don't "load" the image if we want to push it? Why?

Reply:
The action doesn't let you put both `load: true` and `push: true`. It wants them to be mutually exclusive. And it also won't work if both are false :sweat_smile: And also if you omit them both, since both default to false :smile: 

If you don't load when you're not pushing, then the image is built but is not available in your local Docker registry, I think.